CatNote.DAL/Repositories/GenericRepository.cs

```diff
@@ -41,14 +43,23 @@ public async Task<IEnumerable<TEntity>> GetAll(CancellationToken cancellationTok
 
     public async Task<TEntity> GetById(int id, CancellationToken cancellationToken)
     {
-        return await dbSet.AsNoTracking().FirstOrDefaultAsync(x => x.Id == id, cancellationToken);
+        return await dbSet.AsNoTracking().FirstOrDefaultAsync(x => x.Id == id, cancellationToken) ??
+               throw new NotFoundException($"Entity with Id {id} not found.");
     }
 
     public async Task<TEntity> Update(TEntity element, CancellationToken cancellationToken)
     {
-        dbContext.Entry(element).State = EntityState.Modified;
-        await dbContext.SaveChangesAsync(cancellationToken);
+        var existingEntity = await dbContext.Set<TEntity>().FindAsync(element.Id);
 
-        return element;
+        if (existingEntity != null)
```

Comment:
The same as above

Reply:
I think it's not important, because the behaviour shouldn't change whether the field is found in the database or not